### ISSUE TITLE
New instructions needed for PW3 | Must install hotfix twice

### ISSUE THREAD

**Issue by theol0403** [notmarek/LanguageBreak#48]
Hello, thanks for this jailbreak!

The process I had to get it 100% working varied from the posted instructions.

**Existing Behavior**:
* Select Chinese
* Jailbreak logs show up
* **UI is in English, not Chinese**
* ;uzb and copy hotfix + UYK
* **Installing hotfix fails with error + log dump on boot**
* Use demo trick to exit managed mode

Here, I *thought* that everything was properly installed. `;log mrpi` was working, KUAL, was opening, etc. 

But... the behaviour was really strange. `mkk` was not present. The root filesystem was full of permissions errors. Launching USBnetwork or KOReader from KUAL wasn't working at all (and silently failing). Looking at the logs, it was full of permissions errors. 

The solution was to install the hotfix again. Now, mkk is there, and the apps work.

**Expected Behavior**:
It would be great if you could add instructions that specify to install the hotfix twice. I'm not sure if switching from managed mode removed it, but if `mkk` is not present, you need to install the hotfix again. 

Fixes #47 and fixes #46

**Comment by 155c5e** [comment]
thank you so much! worked right away. didn't even occur me to try this.

**Comment by FennekFox19** [comment]
Can you please make a step by step guide edit in this issue to make the work of the programmer a little less painfull?
That would be beneficial for us all because he can link the issue in the readme file for those having the same device and (mostly) firmware (I assume you are on pw3 with fw 5.16.2.1.1) that you have.

Thanks for solving my and many other peoples exact same problem btw.

(little side note: this is a pw3 only issue afaik.)

**Comment by FennekFox19** [comment]
Tested it on the Japan version(32GB model) of the PW3 and the issue isn't occuring there.

Region specific issues?

I might look into that a little bit more and try to rewrite the original jb script or create another method that works for the EU models without a hitch.

**Comment by ElonGaties** [comment]
as a quick note: you dont need to go back into demo mode to reinstall the hotfix, you can just place the hotfix on the device then update

**Comment by Gendozz505** [comment]
Hello

Thanx for the release.

I am wondering where do I find those hotfix .bin files? they are not listed in the repository.\\

Best regards, Genadi.

**Comment by sugar-joh** [comment]
I am on pw3 with fw 5.16.2.1.1 and I have tries reinstalling the hotfix numerous times but the mkk directory does not show up. ;log works though. What should i do?

**Comment by licopper** [comment]
I've got exactly same issue with PW5 SE
And reinstalling update_hotfix once again helped to solve it (partially? not sure) (Update_hotfix_languagebreak-en-GB.bin in my case). I also see KPPMainAppV2_3277_002-juno_16020101_ballatrix-409748_crash dump being created in the 'documents' folder upon every device restart even after reapplying the update_hotfix.

**Comment by snormonster** [comment]
Just a few observations after spending a couple of hours on getting things to work on PW3@5.16.2.1.1

The release zip (v1.0.2.1) that I downloaded from github didn't contain the hotfix, nor `patchedUks.sqsh` inside LanguageBreak - I downloaded the zipped source, but I should have downloaded [LanguageBreak-16.11.23.tar.gz](https://github.com/notmarek/LanguageBreak/releases/download/1.0.2.1/LanguageBreak-16.11.23.tar.gz). (My bad.) It didn't correctly apply the jailbreak. It also did not work when I eventually got the right folder with the Hotfix, the Hotfix kept throwing an error as soon as the Kindle reboots to apply the update from UYK.

Nothing that I tried after that worked anymore. I then Factory reset my Kindle, and re-applied the stock 5.16.2.1.1 firmware via UYK to get a clean slate. 

Then I started from scratch again, going through the steps everything worked mostly as mentioned in the guide, except after I selected Chinese and the jailbreak logs appeared a "Collecting Debug Info" popup appeared. I dismissed it and then was greeted by an "Application Error" popup. Dismissing this left me with a blank screen and nothing happening. I had to hold the power button a few times for a long time, until the device rebooted. I was then able to do ;uzb -> copy hotfix -> ;dsts -> UYK. 

After rebooting, my PW3 wasn't in demo mode anymore, ;log worked, and mkk folder was correctly present etc. 

Thanks to all contributors for this

**Comment by higorslva** [comment]
Same here with Kindle 8th gen. I had to install the hotfix twice and now everything is going well

**Comment by snormonster** [comment]
> @snormonster "The release zip (v1.0.2.1) that I downloaded from github didn't contain the hotfix, nor patchedUks.sqsh inside LanguageBreak" - could you tell me what files are needed? I only have .demo, documents, DONT CHECH BATTERY, jb i in LanguageBreak folder Is patchedUks.sqsh required in files to copy? any others?
> 
> I have encountered Update failed after moving file "Update_hotfix_languagebreak-en-GB.bin" and restarting

Yes, I needed the one that contained patchedUks.sqsh. If you download the release from here, you will have everything that you need: [You have to download [https://github.com/notmarek/LanguageBreak/releases/download/1.0.2.1/LanguageBreak-16.11.23.tar.gz](url)](url)

**Comment by PPrincen87** [comment]
Just to be sure, this should work on a Kindle 10 running on 5.16.21?

**Comment by notmarek** [comment]
No

**Comment by dally-boy** [comment]
> @snormonster "The release zip (v1.0.2.1) that I downloaded from github didn't contain the hotfix, nor patchedUks.sqsh inside LanguageBreak" - could you tell me what files are needed? I only have .demo, documents, DONT CHECH BATTERY, jb i in LanguageBreak folder Is patchedUks.sqsh required in files to copy? any others?
> 
> I have encountered Update failed after moving file "Update_hotfix_languagebreak-en-GB.bin" and restarting

For me I've tried moving and it said "The file name you specified is not valid or too long"
